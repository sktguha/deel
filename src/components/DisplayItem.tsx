
export function DisplayItem({selectedOptions, option, onSelectedOptionsChange, getBoldedText, inputRef}) {
  return (<div
    style={{ cursor: "pointer" }}
    onClick={() => {
      alert(option.label);
      if (selectedOptions.find(sOption => sOption.id === option.id)) {
        return;
      }
      onSelectedOptionsChange([...selectedOptions, option]);
    }} key={option.id}>{getBoldedText(option.label, inputRef.current.value)}</div>);
}
