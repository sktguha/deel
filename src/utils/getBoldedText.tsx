export default function getBoldedText(label, text) {
    return <div dangerouslySetInnerHTML={{
      __html: label.split(text).join("<b>" + text + "</b>")
    }} />;
  }