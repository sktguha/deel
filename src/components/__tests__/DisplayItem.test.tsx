import React from 'react';
import { DisplayItem } from "../DisplayItem"
import { render , screen } from '@testing-library/react';
import getBoldedText from '../../utils/getBoldedText';


test("DisplayItem base",()=>{
    const { container } = render(<DisplayItem getBoldedText={getBoldedText} inputRef={{ current: { value: "te"}}} option={{label: "test", id: 1}}/>)
    expect(container).toMatchSnapshot();
})

test("DisplayItem last item",()=>{
  const { container } = render(<DisplayItem getBoldedText={getBoldedText} isLast inputRef={{ current: { value: "te"}}} option={{label: "test", id: 1}}/>)
  expect(container).toMatchSnapshot();
})

test("DisplayItem multiple options",()=>{
  const { container } = render(<DisplayItem getBoldedText={getBoldedText} isLast inputRef={{ current: { value: "te"}}} 
  option={{label: "test2", id: 2}}/>)
  expect(container).toMatchSnapshot();
})

test("DisplayItem multiple options last item",()=>{
  const { container } = render(<DisplayItem isLast getBoldedText={getBoldedText} isLast inputRef={{ current: { value: "te"}}} 
  option={{label: "test2", id: 2}}/>)
  expect(container).toMatchSnapshot();
})
