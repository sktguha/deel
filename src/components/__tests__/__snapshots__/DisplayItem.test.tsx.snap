// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DisplayItem base 1`] = `
<div>
  <div
    class="menu-item undefined"
    style="cursor: pointer;"
  >
    <div>
      <b>
        te
      </b>
      st
    </div>
  </div>
</div>
`;

exports[`DisplayItem last item 1`] = `
<div>
  <div
    class="menu-item menu-item-last"
    style="cursor: pointer;"
  >
    <div>
      <b>
        te
      </b>
      st
    </div>
  </div>
</div>
`;

exports[`DisplayItem multiple options 1`] = `
<div>
  <div
    class="menu-item menu-item-last"
    style="cursor: pointer;"
  >
    <div>
      <b>
        te
      </b>
      st2
    </div>
  </div>
</div>
`;

exports[`DisplayItem multiple options last item 1`] = `
<div>
  <div
    class="menu-item menu-item-last"
    style="cursor: pointer;"
  >
    <div>
      <b>
        te
      </b>
      st2
    </div>
  </div>
</div>
`;
